src/main/java/pvws/servlets/PVGetServlet.java

```diff
@@ -0,0 +1,369 @@
+
+    /**
+     * Handles GET requests to /pvget and writes the response in JSON format.
+     * 
+     * The method supports two input formats:
+     * - A single PV name passed directly via the "pv" parameter.
+     * - Multiple PV names passed as a JSON array of strings via the "pv" parameter.
+     * 
+     * @param request The HTTP request containing the "pv" parameter specifying one or more PVs.
+     * @param g       The JsonGenerator used to write the JSON response.
+     * @throws IOException If an I/O error occurs while writing the JSON response.
+     */
+    @Override
+    protected void writeJson(final HttpServletRequest request, final JsonGenerator g) throws IOException
+    {
+        // Retrieve the "pv" parameter from the request, which specifies the PV(s) to fetch.
+        String pvParam = request.getParameter("pv");
+
+        // If no PVs are specified, return an error response.
+        if (pvParam == null || pvParam.isEmpty()) {
+            g.writeStartObject();
+            g.writeStringField("name", pvParam);
+            g.writeBooleanField("success", false);
+            g.writeStringField("message", "No PVs have been specified.");
+            g.writeEndObject();
+            return;
+        }
+
+        List<String> pvs = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper(json_factory);
+
+        try {
+            // Try parsing as JSON array
+            JsonNode node = mapper.readTree(pvParam);
+            if (node.isArray()) {
+                for (JsonNode item : node) {
+                    pvs.add(item.asText());
+                }
+            } else {
+                // Not an array, assume it's a single PV name
+                pvs.add(pvParam);
+            }
+        } catch (JsonProcessingException e) {
+            // Invalid JSON, treat it as a single PV name
+            pvs.add(pvParam);
+        }
+        
+        // Handle single PV request.
+        if (pvs.size() == 1) {
+            try {
+                // Generate a JSON object for the single PV and write it directly to the response.
+                String jsonObject = generateJsonForPV(pvs.get(0));
+                g.writeRawValue(jsonObject); // Write raw JSON string
+            } catch (Exception e) {
+                // If an exception occurs, return an error response for the single PV.
+                g.writeStartObject();
+                g.writeStringField("name", pvs.get(0));
+                g.writeBooleanField("success", false);
+                g.writeStringField("message", "An exception occurs when fetching or formatting PV value.");
+                g.writeEndObject();
+            }
+        } else {
+            // Handle multiple PV requests using a thread pool for concurrent processing.
+            int numberOfThreads = 100; // Set a large number of threads for I/O-bound tasks.
+            ExecutorService executorService = Executors.newFixedThreadPool(numberOfThreads);
```

Comment:
Are you creating a new thread pool for 100 threads each time the servlet is called?!?